discover/deployment/articles-dxp/01-installing-search-engine/06-xpack-security.markdown

```diff
@@ -0,0 +1,240 @@
+
+    This generates another ZIP file. Extract the contents somewhere in the
+    `Elasticsearch Home/config` folder.
+
++$$$
+
+**Note:** The `certutil` command defaults to using the *PKSC#12* format
+for certificate generation. Kibana does not work with PKSC#12
+certificates, so the `--pem` option (to generate the certificate in PEM
+format) is important if you're using X-Pack monitoring.
+
+$$$
+
+### Enable TLS [](id=enable-tls)
+
+[Enable TLS](https://www.elastic.co/guide/en/elasticsearch/reference/6.1/configuring-tls.html#enable-ssl) 
+on each node via its `elasticsearch.yml`.
+
+1.  Add the certificate, key and certificate authority paths to each node's
+    `elasticsearch.yml`:
+
+        xpack.ssl.certificate: /path/to/[Elasticsearch Home]/config/localhost.crt
+        xpack.ssl.key: /path/to/[Elasticsearch Home]/config/localhost.key
+        xpack.ssl.certificate_authorities: ["/path/to/ca.crt"]
+
+    The example paths above assume you added the certificate to `Elasticsearch
+    Home/config/`. 
+
+2.  Enable transport layer TLS with these settings in `elasticsearch.yml`:
+
+        xpack.security.transport.ssl.enabled: true
+        xpack.security.transport.ssl.verification_mode: certificate
+
+3.  Enable TLS on the HTTP layer to encrypt client communication:
+
+        xpack.security.http.ssl.enabled: true
+
+After X-Pack is installed and TLS is enabled, configure the X-Pack Security
+adapter in @product@.
+
+## Install and Configure the Liferay Connector to X-Pack Security [](id=install-and-configure-the-liferay-connector-to-x-pack-security)
+
+If you have a Liferay Enterprise Search Premium subscription, 
+[download](https://web.liferay.com/group/customer/dxp/downloads/enterprise-search) 
+the Liferay Connector to X-Pack Security [Elastic Stack 6.x]. Install the LPKG
+file by copying it into the `Liferay Home/deploy` folder. That's all there is to
+it.
+
+To configure the X-Pack adapter, navigate to *Control Panel*
+&rarr; *Configuration* &rarr; *System Settings*. Find the *Foundation* category and
+click on the *X-Pack Security* entry. You can enter the property values here, but
+it's more common to use a 
+[configuration file](/discover/portal/-/knowledge_base/7-0/understanding-system-configuration-files)
+deployed to `Liferay Home/osgi/configs`. For the X-Pack adapter, create a file
+called
+
+    com.liferay.portal.search.elasticsearch6.xpack.security.internal.configuration.XPackSecurityConfiguration.config
+
+The exact contents of the file depend on your X-Pack setup. To configure the adapter
+according to the Elasticsearch setup documented here, populate the file with
+these contents:
+
+    sslKeyPath="/path/to/[Elasticsearch Home]/config/localhost.key"
+    sslCertificatePath="/path/to/[Elasticsearch Home]/config/localhost.crt"
+    certificateFormat="PEM"
+    requiresAuthentication="true"
+    username="elastic"
+    password="GqhoaEUyTM@tp1*wQd~F"
```

Comment:
I changed to the interactive password setup, but never updated this use of the autogenerated password. Change to _liferay_

Reply:
also added to text: 

>Note that the `password` should match what you set during the X-Pack password setup above.